specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-06-01/Marketplace.json

```diff
@@ -2039,6 +2129,115 @@
         }
       }
     },
+    "BulkCollectionsPayload": {
+      "description": "Bulk collections action properties",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BulkCollectionsDetails",
+          "description": "bulk collections properties details",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "BulkCollectionsDetails": {
+      "description": "Bulk collection details",
+      "type": "object",
+      "properties": {
+        "collectionIds": {
+          "description": "collection ids list that the action is performed on",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "action": {
+          "description": "Action to perform (For example: EnableCollections, DisableCollections)",
+          "type": "string"
+        }
+      }
+    },
+    "BulkCollectionsResponse": {
+      "description": "The bulk collections response. The response contains two lists that indicate for each collection whether the operation succeeded or failed",
+      "type": "object",
+      "properties": {
+        "succeeded": {
+          "description": "Succeeded collections",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CollectionsDetails"
+          }
+        },
+        "failed": {
+          "description": "Failed collections",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CollectionsDetails"
+          }
+        }
+      }
+    },
+    "QueryApprovedPlansPayload": {
+      "description": "Query approved plans payload",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/QueryApprovedPlans",
+          "description": "Query approved plans details",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "QueryApprovedPlans": {
+      "description": "Query approved plans details",
+      "type": "object",
+      "properties": {
+        "offerId": {
+          "description": "Offer id",
+          "type": "string"
+        },
+        "planIds": {
+          "description": "Offer plan ids",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "QueryApprovedPlansResponse": {
```

Comment:
This will be easier to consume if its an array of plan objects with planId as a property in each object. Why does it need to be a dictionary with dynamic keys?

Reply:
We want to enforce the plan id uniqueness in the result

Reply:
You control the response so you can enforce uniqueness regardless of the response schema.  If that is the only reason please use a strict schema instead of dynamic keys

Reply:
Fixed to list of objects